launchpod/src/main/java/com/google/launchpod/servlets/FormHandlerServlet.java

```diff
@@ -0,0 +1,125 @@
+  public static final String EMAIL = "email";
+  public static final String TIMESTAMP = "timestamp";
+  public static final String MP3LINK = "mp3link";
+  public static final String XML_STRING = "xmlString";
+  public static final String PUB_DATE= "pubDate";
+
+  private static final String ID = "id";
+
+  public static final Gson GSON = new Gson();
+
+  /**
+   * request user inputs in form fields then create Entity and place in datastore
+   */
+  @Override
+  public void doPost(HttpServletRequest req, HttpServletResponse res) throws IOException {
+    String podcastTitle = req.getParameter(PODCAST_TITLE);
+    String mp3Link = req.getParameter(MP3LINK);
+    if((podcastTitle.isEmpty() || podcastTitle == null) || (mp3Link.isEmpty() || mp3Link == null)){
+      throw new IOException("No Title or MP3 link inputted, please try again.");
+    }
+    long timestamp = System.currentTimeMillis();
+    // Create time
+    DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("yyyy/MM/dd HH:mm:ss");
+    LocalDateTime publishTime = LocalDateTime.now();
+    String pubDate = dateFormatter.format(publishTime);
+
+    //Create entity with all desired attributes
+    Entity userFeedEntity = new Entity(USER_FEED);
+    userFeedEntity.setProperty(PODCAST_TITLE, podcastTitle);
+    userFeedEntity.setProperty(MP3LINK, mp3Link);
+    userFeedEntity.setProperty(TIMESTAMP, timestamp);
+    userFeedEntity.setProperty(PUB_DATE, pubDate);
+
+    // Generate xml string
+    String xmlString = "";
+    try{
+      xmlString = xmlString(userFeedEntity);
+      userFeedEntity.setProperty(XML_STRING ,xmlString);
+    }catch(IOException e){
+      throw new IOException("Unable to create XML string.");
+    }
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(userFeedEntity);
+
+    //return accessible link to user 
+    String urlID = KeyFactory.keyToString(userFeedEntity.getKey());
+    String rssLink = "https://launchpod-step18-2020.appspot.com/rss-feed?id=" + urlID;
+    res.setContentType("text/html");
+    res.getWriter().println("<a href=\"" + rssLink + "\">" + rssLink + "</a>");
+  }
+
+  /**
+  * Display RSS feed xml string that user tries recalling  with the given ID
+  */
+  @Override
+  public void doGet(HttpServletRequest req, HttpServletResponse res) throws IOException {
+    // Get ID passed in request
+    String id = req.getParameter(ID);
+    Key urlID = KeyFactory.stringToKey(id);
+    // Search key in datastore
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    // create entity that contains id from datastore
+    Entity desiredFeedEntity;
+    try {
+      desiredFeedEntity = datastore.get(urlID);
+      // generate xml string
+      UserFeed desiredUserFeed = UserFeed.fromEntity(desiredFeedEntity);
+      RSS rssFeed = new RSS(desiredUserFeed.getTitle(), desiredUserFeed.getLink(), desiredUserFeed.getPubDate());
```

Comment:
can you double check that desiredUserFeed.getTitle() and the other fields aren't empty?

Reply:
Just checked, they do contain the input from the form